pandas-stubs/core/indexes/base.pyi

```diff
@@ -223,7 +222,7 @@ class Index(IndexOpsMixin, PandasObject):
     def shape(self) -> tuple[int, ...]: ...
     # Extra methods from old stubs
     def __eq__(self, other: object) -> np_ndarray_bool: ...  # type: ignore[override]
-    def __iter__(self) -> Iterator[IndexIterScalar | tuple[Hashable, ...]]: ...
+    def __iter__(self) -> Iterator: ...
```

Comment:
If we were to type the return type, shouldn't this simply have been `Iterator[Hashable]` (tuple and everything in IndexIterScalar) is hashable?

(Having mypy_primer could have been helpful to see the false-positives earlier.)

Reply:
https://github.com/pandas-dev/pandas-stubs/pull/367#issuecomment-1304724134

Reply:
> If we were to type the return type, shouldn't this simply have been `Iterator[Hashable]` (tuple and everything in IndexIterScalar) is hashable?
> 
> (Having mypy_primer could have been helpful to see the false-positives earlier.)

If I change it to `Iterator[Hashable]`, then I get this with the test (and example above) from pyright:
```text
Operator "in" not supported for types "Literal['at']" and "Hashable"
  Operator "in" not supported for types "Literal['at']" and "Hashable"
```